_posts/2020-02-7-dns-based-discovery.md

```diff
@@ -0,0 +1,94 @@
+---
+layout: post
+name:  "DNS based discovery"
+title:  "DNS based discovery"
+date:   2020-02-7 12:00:00 +0100
+author: decanus
+published: true
+permalink: /dns-based-discovery
+categories: research
+summary: A look at EIP-1459 and the benefits of DNS based discovery.
+image: /assets/img/vac.jpg
+---
+
+Discovery in p2p networks is the process of how nodes find each other and specific resources they are looking for. Popular discovery protocols, such as [Kademlia](https://pdos.csail.mit.edu/~petar/papers/maymounkov-kademlia-lncs.pdf) which utilizes a [distributed hash table](https://en.wikipedia.org/wiki/Distributed_hash_table) or DHT, are highly inefficient for resource restricted devices, as we cannot expect a mobile phone for example to synchronize an entire DHT using cellular data.
```

Comment:
> as we cannot expect a mobile phone for example to synchronize an entire DHT using cellular data.

That's part of it, but I think a bigger issue is in terms of connection windows being too short and it being battery intensive to keep establishing connections, etc. I'd emphasize those two more than the cellular data part